lib/error_reporter.h

```diff
@@ -91,11 +91,12 @@ class ErrorReporter {
     }
 
     // error message for a bug
-    template <typename... T>
-    std::string bug_message(const char *format, T... args) {
+    template <typename... Args>
+    std::string bug_message(const char *format, Args &&...args) {
```

Comment:
missing inluce `bug_helper.h`, or maybe this function could be elsewhere (in `exceptions.h`)?

Reply:
Added, yes, was leaking via other headers